src/storage/txn/commands/atomic_store.rs

```diff
@@ -52,14 +52,21 @@ impl<S: Snapshot, L: LockManager> WriteCommand<S, L> for RawAtomicStore {
         let mut data = vec![];
         let rows = self.mutations.len();
         let (cf, mutations, ctx) = (self.cf, self.mutations, self.ctx);
-        let expire_ts = self.ttl.map(convert_to_expire_ts);
+        let expire_ts: Vec<Option<u64>> = self
+            .ttls
+            .unwrap()
+            .iter()
+            .map(|ttl| Some(convert_to_expire_ts(*ttl)))
+            .collect();
+        let mut idx = 0;
         for m in mutations {
             match m {
                 Mutation::Put((key, value)) => {
                     let mut m = Modify::Put(cf, key, value);
-                    if let Some(ts) = expire_ts {
+                    if let Some(ts) = expire_ts[idx] {
                         m.with_ttl(ts);
                     }
+                    idx += 1;
```

Comment:
ditto

Reply:
this function uses "for loop" to traverse kvs, so I think it is no need to zip iter

Reply:
Indexing is evil. By using zip, you can `for loop` the kvs together with ttls